test/integration_test/basic_test.go

```diff
@@ -771,3 +788,59 @@ func AlertManagerRegression(tc *types.TestCase) func(*testing.T) {
 		ci_utils.UninstallAndValidateStorageCluster(cluster, t)
 	}
 }
+
+// InstallWithNodeTopologyLabels includes the following steps:
+// 1. label all k8s nodes with topology key 'topology.kubernetes.io/region' and different values
+// 2. install StorageCluster with stork, csi, pvc controller enabled
+// 3. validate topology keys from deployment.spec.template.spec.topologySpreadConstraints matches the topology labels
+// 4. add a new label 'topology.kubernetes.io/zone' to nodes, then validate new key should be added to deployments
+// 5. remove all topology labels, then do step 3 again to validate all topology keys should be removed
+// 6. uninstall the cluster
+func InstallWithNodeTopologyLabels(tc *types.TestCase) func(*testing.T) {
+	return func(t *testing.T) {
+		// Add topology label to all nodes
+		nodeList, err := coreops.Instance().GetNodes()
+		require.NoError(t, err)
+		regionKey := "topology.kubernetes.io/region"
+		topologyKeys := []string{regionKey}
+		for i, node := range nodeList.Items {
```

Comment:
I would say copy the original `topology.kuberenetes.io/*` values and remove them before we overwrite with our own values. Once we start deploying on clouds, it would have some values already set, which make this test behave unexpectedly. Reset the original values back when the test is successful.